src/text_engine/boundaries/Door.java

```diff
@@ -52,21 +52,22 @@ public Room getRoom2() {
      * @throws IllegalStateException door is locked
      * @throws IllegalArgumentException given room is not connected to this door
      */
-    public Room getOtherRoom(Room from) {
+    public Room getOtherRoom(Room from) throws IllegalArgumentException, IllegalStateException {
         Objects.requireNonNull(from);
```

Comment:
Pretty sure I was the one to drop these around, but for the sake of consistency, do we want to exclusively use @NotNull in the params for this?